### PR TITLE
add support for trailing commas in ADT constructors and function arg lists

### PR DESCRIPTION
**Pull request by bmakuh**
Addresses remainder of #182

### CODE REVIEW

**Review by ospencer — APPROVED** [review]
Awesome! So cool to finally be able to close this issue 😄

File: compiler/src/parsing/parser.dyp [hunk 1]
```diff
@@ -523,7 +523,7 @@ lam_args :
   | patterns? { Option.value ~default:[] $1 }
 
 lam_expr :
-  | lparen lam_args rparen thickarrow block_or_expr { Exp.lambda ~loc:(symbol_rloc dyp) $2 $5 }
+  | lparen lam_args comma? rparen thickarrow block_or_expr { Exp.lambda ~loc:(symbol_rloc dyp) $2 $5 }
```
Comment (ospencer) [inline compiler/src/parsing/parser.dyp:526]: Looks like this is why the CI is failing.
```suggestion
  | lparen lam_args comma? rparen thickarrow block_or_expr { Exp.lambda ~loc:(symbol_rloc dyp) $2 $6 }
```

Comment (bmakuh) [reply compiler/src/parsing/parser.dyp]: Oops! 🙈 Let me fix that.

Comment (ospencer) [reply compiler/src/parsing/parser.dyp]: I went ahead and committed it!

**Comment by ospencer** [comment]
@bmakuh Looks like your tests are good, and you just need to run `yarn compiler format` now 👍

**Comment by bmakuh** [comment]
@ospencer ok, I ran the formatter. 🤞 should pass now

**Comment by ospencer** [comment]
Sweet, thanks for your hard work on this!

**Comment by bmakuh** [comment]
Woo! Glad I could help!

**Pull request merged**